Code/GraphMol/FileParsers/MolSupplier.h

```diff
@@ -65,6 +65,14 @@ class RDKIT_FILEPARSERS_EXPORT MolSupplier {
   virtual bool atEnd() = 0;
   virtual ROMol *next() = 0;
 
+  virtual void close() {
+    if (df_owner && dp_inStream) {
```

Comment:
```suggestion
    if (df_owner) {
```
`delete` will do nothing if `dp_inStream` is `nullptr`, no need to check (indeed there is no check further on in `SmilesWriter::close`)

Reply:
yeah, that check was old. Might as well fix it.